source/common/http/http1/codec_impl.cc

```diff
@@ -483,10 +515,19 @@ void ConnectionImpl::onHeaderValue(const char* data, size_t length) {
     throw CodecProtocolException("http/1.1 protocol error: header value contains NUL");
   }
 
-  header_parsing_state_ = HeaderParsingState::Value;
+  if (processing_trailers_) {
+    trailer_parsing_state_ = HeaderParsingState::Value;
+  } else {
+    header_parsing_state_ = HeaderParsingState::Value;
+  }
   current_header_value_.append(data, length);
 
+  validateHeaderMapSize();
+}
+
+void ConnectionImpl::validateHeaderMapSize() {
   // Verify that the cached value in byte size exists.
+  ASSERT(current_header_map_.get() != nullptr, "current_header_map_ is nullptr");
```

Comment:
nit: del, this will crash in an obvious way on the next line, I would just delete.